packages/menu-surface/index.tsx

```diff
@@ -347,6 +350,7 @@ class MenuSurface extends React.Component<MenuSurfaceProps, MenuSurfaceState> {
       children,
       ...otherProps
     } = this.props;
+    if (!document) return null;
```

Comment:
@ben-mckernan yes please use a mounted flag as described [here](https://github.com/facebook/react/issues/13097#issuecomment-405658104). Please copy the pattern set in [linear-progress](https://github.com/material-components/material-components-web-react/blob/rc0.11.0/packages/linear-progress/index.tsx#L47). I don't think the property needs to be on state unless you can think of one.

Reply:
It needs to be on state otherwise a re-render will not be triggered. The initial render will be null since it runs before `componentDidMount` is called for the first time.